canal/sync.go

```diff
@@ -2,10 +2,12 @@ package canal
 
 import (
 	"fmt"
-	"regexp"
 	"time"
 
 	"github.com/pingcap/errors"
+	"github.com/pingcap/parser"
+	"github.com/pingcap/parser/ast"
+	_ "github.com/pingcap/tidb/types/parser_driver"
```

Comment:
IMO, it is bad to import tidb repo, too big.

we should only import parser repo.

Reply:
fine, i had remove diver import and use an empty impl of expr to avoid panic of check in parser.New()